app/kuma-dp/pkg/dataplane/meshmetrics/component.go

```diff
@@ -0,0 +1,133 @@
+package meshmetrics
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/app/kuma-dp/pkg/dataplane/metrics"
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+	"github.com/kumahq/kuma/pkg/plugins/policies/meshmetric/plugin/xds"
+	utilnet "github.com/kumahq/kuma/pkg/util/net"
+)
+
+type ConfigFetcher struct {
+	httpClient     http.Client
+	socketPath     string
+	ticker         *time.Ticker
+	hijacker       *metrics.Hijacker
+	defaultAddress string
+	envoyAdminPort uint32
+}
+
+var _ component.Component = &ConfigFetcher{}
+
+var logger = core.Log.WithName("mesh-metric-config-fetcher")
+
+func NewMeshMetricConfigFetcher(socketPath string, ticker *time.Ticker, hijacker *metrics.Hijacker, address string, envoyAdminPort uint32) component.Component {
+	return &ConfigFetcher{
+		httpClient: http.Client{
+			Transport: &http.Transport{
+				DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+					return net.Dial("unix", socketPath)
+				},
+			},
+		},
+		socketPath:     socketPath,
+		ticker:         ticker,
+		hijacker:       hijacker,
+		defaultAddress: address,
+		envoyAdminPort: envoyAdminPort,
+	}
+}
+
+func (cf *ConfigFetcher) Start(stop <-chan struct{}) error {
+	logger.Info("starting Dynamic Mesh Metrics Configuration Scraper",
+		"socketPath", fmt.Sprintf("unix://%s", cf.socketPath),
+	)
+
+	for {
+		select {
+		case <-cf.ticker.C:
+			if _, err := os.Stat(cf.socketPath); errors.Is(err, os.ErrNotExist) {
+				logger.V(1).Info("skipping /meshmetric endpoint scrape since socket does not exist", "err", err)
+				continue
+			}
+
+			configuration, err := cf.scrapeConfig()
+			if err != nil {
+				continue
+			}
+			logger.V(1).Info("updating hijacker configuration", "conf", configuration)
+			cf.hijacker.SetApplicationsToScrape(cf.mapApplicationToApplicationToScrape(configuration.Observability.Metrics.Applications))
+		case <-stop:
+			logger.Info("stopping Dynamic Mesh Metrics Configuration Scraper")
+			return nil
+		}
+	}
+}
+
+func (cf *ConfigFetcher) NeedLeaderElection() bool {
+	return false
+}
+
+func (cf *ConfigFetcher) scrapeConfig() (*xds.MeshMetricDpConfig, error) {
+	configuration, err := cf.httpClient.Get("http://localhost/meshmetric")
+	if err != nil {
+		logger.Info("failed to scrape /meshmetric endpoint", "err", err)
+		return nil, err
```

Comment:
if we want to follow the madr https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/003-logging.md?plain=1#L56 we should either log or return the error.

So this should be something like:

```suggestion
		return nil, errors.wrap("failed to scrape /meshmetric endpoint", err)
```

and in line 67 log

Reply:
changed